training/src/anemoi/training/diagnostics/callbacks/plot.py

```diff
@@ -670,16 +700,27 @@ def __init__(self, config: OmegaConf, every_n_epochs: int | None = None) -> None
         """
         super().__init__(config, every_n_epochs=every_n_epochs)
         self.q_extreme_limit = config.get("quantile_edges_to_represent", 0.05)
+        self.dataset_names = dataset_names if dataset_names is not None else ["data"]
 
-    def get_node_trainable_tensors(self, node_attributes: NamedNodesAttributes) -> dict[str, torch.Tensor]:
+    def get_node_trainable_tensors(
+        self,
+        node_attributes: NamedNodesAttributes,
+        dataset_name: str,
+    ) -> dict[str, torch.Tensor]:
         return {
-            name: tt.trainable for name, tt in node_attributes.trainable_tensors.items() if tt.trainable is not None
+            name: tt.trainable
+            for name, tt in node_attributes[dataset_name].trainable_tensors.items()
+            if tt.trainable is not None
         }
 
-    def get_edge_trainable_modules(self, model: torch.nn.Module) -> dict[tuple[str, str], torch.Tensor]:
+    def get_edge_trainable_modules(
+        self,
+        model: torch.nn.Module,
+        dataset_name: str,
+    ) -> dict[tuple[str, str], torch.Tensor]:
         trainable_modules = {
-            (model._graph_name_data, model._graph_name_hidden): model.encoder,
-            (model._graph_name_hidden, model._graph_name_data): model.decoder,
+            (model._graph_name_data, model._graph_name_hidden): model.encoder[dataset_name],
```

Comment:
_graph_name_hidden I understand that if we have one processor would be the same - but for _graph_name_data I would have imagined this would also be dependant of the dataset as in each encoder would have a different input graph if they use different datasets?